### PR TITLE
set up database in travis for the testsuite

### PR DESCRIPTION
**Pull request by tampakrap**

### CODE REVIEW

**Comment by smithfarm** [comment]
I don't think REST will work with database set up in this way. The proper way to create and set up the database is to run the `bin/dochazka-resetdb` script.

**Comment by tampakrap** [comment]
It's still WIP, I'll let you know when it is ready

**Comment by smithfarm** [comment]
ah, ok - please prefix wip PRs with "DNM" (Do Not Merge). I did it for you this time :-)

File: .travis.yml [hunk 1]
```diff
@@ -3,4 +3,16 @@ language: perl
...
+addons:
+  postgresql: '9.4'
+before_script:
+  - sudo mkdir /etc/dochazka-rest
+  - sudo sh -c "echo \"set( 'DOCHAZKA_DBNAME', 'dochazka-test' );\" >> /etc/dochazka-rest/REST_SiteConfig.pm"
+  - sudo sh -c "echo \"set( 'DOCHAZKA_DBUSER', 'dochazka' );\" >> /etc/dochazka-rest/REST_SiteConfig.pm"
+  - sudo sh -c "echo \"set( 'DOCHAZKA_DBPASS', 'dochazka' );\" >> /etc/dochazka-rest/REST_SiteConfig.pm"
+  - chmod +x bin/*
+  - sed -i -e 's#\(.*dbinit\)#bin/\1#' bin/dochazka-resetdb
+  - sudo bin/dochazka-resetdb
```
Comment (smithfarm) [inline .travis.yml]: Rebase to latest master and then:

```
sudo sh bin/dochazka-resetdb
sudo perl bin/dochazka-dbinit
```

File: .travis.yml [hunk 2]
```diff
@@ -3,4 +3,16 @@ language: perl
...
+  - postgresql
+addons:
+  postgresql: '9.4'
+before_script:
+  - sudo mkdir /etc/dochazka-rest
+  - sudo sh -c "echo \"set( 'DOCHAZKA_DBNAME', 'dochazka-test' );\" >> /etc/dochazka-rest/REST_SiteConfig.pm"
+  - sudo sh -c "echo \"set( 'DOCHAZKA_DBUSER', 'dochazka' );\" >> /etc/dochazka-rest/REST_SiteConfig.pm"
+  - sudo sh -c "echo \"set( 'DOCHAZKA_DBPASS', 'dochazka' );\" >> /etc/dochazka-rest/REST_SiteConfig.pm"
+  - chmod +x bin/*
+  - sed -i -e 's#\(.*dbinit\)#bin/\1#' bin/dochazka-resetdb
```
Comment (smithfarm) [inline .travis.yml]: Rebase to latest master and drop the `sed`.

File: .travis.yml [hunk 3]
```diff
@@ -3,4 +3,16 @@ language: perl
...
+services:
+  - postgresql
+addons:
+  postgresql: '9.4'
+before_script:
+  - sudo mkdir /etc/dochazka-rest
+  - sudo sh -c "echo \"set( 'DOCHAZKA_DBNAME', 'dochazka-test' );\" >> /etc/dochazka-rest/REST_SiteConfig.pm"
+  - sudo sh -c "echo \"set( 'DOCHAZKA_DBUSER', 'dochazka' );\" >> /etc/dochazka-rest/REST_SiteConfig.pm"
+  - sudo sh -c "echo \"set( 'DOCHAZKA_DBPASS', 'dochazka' );\" >> /etc/dochazka-rest/REST_SiteConfig.pm"
+  - chmod +x bin/*
```
Comment (smithfarm) [inline .travis.yml]: This is not necessary; just "sudo sh $SCRIPT" or "sudo perl $SCRIPT"

Comment (tampakrap) [inline .travis.yml]: why not to make the scripts in bin/ executable in git instead?

Comment (smithfarm) [inline .travis.yml]: I don't see any reason to do that. You're supposed to set the executable bit at install time, in the packaging.

Comment (tampakrap) [inline .travis.yml]: for deployments via git (testing or production at the tag). Modern web frameworks like rails do have executable scripts in bin/ as well

**Comment by smithfarm** [comment]
OK, executable bits added: https://github.com/smithfarm/dochazka-rest/commit/c7b21c4154fa0b248be2863e6c104188b3fe838c

**Comment by tampakrap** [comment]
ready for review

**Pull request merged**